fix: improve output handling for headless mode

gormanity opened the pull request:
This commit fixes an issue when mason-update-all is called in Neovim headles mode (nvim --headless) where each outputed message is appended to the last with no line break in between.

gormanity committed:
fix: improve output handling for headless mode

This commit fixes an issue when mason-update-all is called in Neovim
headles mode (nvim --headless) where each outputed message is appended
to the last with no line break in between. It uses `vim.api.nvim_list_uis()` to
detect when in headless mode so interactive mode is not affected.

RubixDev commented:
I believe I tried something like this before, but it has the issue that it breaks the printing in the normal GUI when a user wants to manually use the `:MasonUpdateAll` command. If you can get that working at the same time, I'd happily merge this

gormanity commented:
I’ve updated the implementation to support both headless and interactive use. Output goes to io.stdout in headless mode, and print() in the GUI. Confirmed it works from both nvim --headless and :MasonUpdateAll. Let me know if you’d like further tweaks!

gormanity committed:
feat: extend print_message to support vim.notify in interactive mode

Builds on the message routing introduced in the previous commit by
adding support for vim.notify when running interactively and the
function is available. This improves the UX of :MasonUpdateAll in Neovim GUIs.

Streamlines message generation by moving the "[mason-update-all]" prefix
into the print_message function for headless and print() modes.
In vim.notify mode, the context is included in the notification title
instead of the message body.

Also addresses the intent of PR #4.

gormanity commented:
I went ahead and updated the implementation in a new commit to extend the print_message selector function to support vim.notify in interactive mode, if it's available. This should also address #4.

RubixDev commented:
lgtm, thanks
